app/src/main/kotlin/com/wire/android/notification/CallNotificationManager.kt

```diff
@@ -0,0 +1,108 @@
+package com.wire.android.notification
+
+import android.app.Notification
+import android.app.PendingIntent
+import android.content.Context
+import android.os.Build
+import androidx.core.app.NotificationChannelCompat
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import com.wire.android.R
+import com.wire.kalium.logic.data.conversation.Conversation
+import com.wire.kalium.logic.data.id.QualifiedID
+import com.wire.kalium.logic.feature.call.Call
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class CallNotificationManager @Inject constructor(private val context: Context) {
+
+    private val notificationManager = NotificationManagerCompat.from(context)
+
+    fun handleNotifications(calls: List<Call>, userId: QualifiedID?) {
+        if (calls.isEmpty() || userId == null) hideCallNotification()
+        else showIncomingCallNotification(calls.first(), userId)
+    }
+
+    fun hideCallNotification() = notificationManager.cancel(NOTIFICATION_ID)
+
+    private fun showIncomingCallNotification(call: Call, userId: QualifiedID) {
+        createNotificationChannelIfNeeded()
+
+        notificationManager.notify(NOTIFICATION_ID, getNotification(call, userId))
+    }
+
+    private fun createNotificationChannelIfNeeded() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val notificationChannel = NotificationChannelCompat
+                .Builder(CHANNEL_ID, NotificationManagerCompat.IMPORTANCE_MAX)
+                .setName(CHANNEL_NAME)
+                .build()
+
+            notificationManager.createNotificationChannel(notificationChannel)
+        }
+    }
+
+    private fun getNotification(call: Call, userId: QualifiedID): Notification {
+        val conversationIdString = call.conversationId.toString()
+        val userIdString = userId.toString()
+        val title = getNotificationTitle(call)
+        val content = getNotificationBody(call)
+
+        return NotificationCompat.Builder(context, CHANNEL_ID)
+            .setContentTitle(title)
+            .setContentText(content)
+            .addAction(getDeclineCallAction(conversationIdString, userIdString))
```

Comment:
Would this also be used when swiping and dismissing the call? or only if user clicks on hang up/reject call button?

Reply:
??

Reply:
good point, i'll add `DeleteIntent`